sklearn/model_selection/_split.py

```diff
@@ -560,9 +569,9 @@ def __init__(self, n_splits=3, shuffle=False, random_state=None):
 
     def _make_test_folds(self, X, y=None, groups=None):
         if self.shuffle:
-            rng = check_random_state(self.random_state)
+            rng = check_random_state(self._random_state)
```

Comment:
I thought about not having this but assumed it would be cleaner to reproduce the input that the user gave in repr instead of random. But your legitimate reasons justify the removal of this... I'll do and push in a while... Thnx